### DISCUSSION TITLE
Is it too annoying to add `isfinite()` preconditions?

### DISCUSSION THREAD

**Discussion by pschanely** [pschanely/CrossHair#266] [General]
As of v0.0.58, CrossHair can generate `math.nan`, `math.inf`, and `-math.inf` as float arguments.

The only officially supported way to avoid this is to add `math.isfinite()` preconditions. But there might be other options.

I'm slightly -1 on having global or broad configuration to disable this behavior - it hurts readability and reproduce-ability to have hidden state that changes the meaning of the contract or test.

If some day, CrossHair understands [annotated-types](https://github.com/annotated-types/annotated-types), then you could write this in a less verbose way as part of the type annotation, e.g.: `def foo(x: IsFinite[float]):`

Who has other options or ideas?

**Comment by andrew222651** [comment]
I think math.isfinite() in preconditions is fine

**Reply by pschanely** [reply]
Thanks for the vote!!
